x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/components/step_edit_hosts.test.tsx

```diff
@@ -111,18 +113,18 @@ describe('StepEditHosts', () => {
     testRenderer = createFleetTestRendererMock();
   });
 
-  it('should display create form when no agent policies', () => {
+  it('should display create form when no agent policies', async () => {
     (useGetAgentPolicies as jest.MockedFunction<any>).mockReturnValue({
       data: {
         items: [],
       },
     });
 
+    (useAllNonManagedAgentPolicies as jest.MockedFunction<any>).mockReturnValue([]);
+
     render();
 
-    waitFor(() => {
-      expect(renderResult.getByText('Agent policy 1')).toBeInTheDocument();
-    });
```

Comment:
`waitFor` wasn't really needed here, and this assertion was skipped since it wasn't waiting for the promise to resolve.